typegen/unification/filter_base.py

```diff
@@ -52,7 +52,5 @@ def apply(self, trace_data: pd.DataFrame) -> pd.DataFrame:
 
         @param trace_data The provided trace data to process.
         """
-        for trace_data_filter in self.filters:
-            trace_data = trace_data_filter.apply(trace_data)
-
-        return trace_data.copy().reset_index(drop=True)
+        applied = functools.reduce(lambda acc, f: f.apply(acc), self.filters, trace_data)
```

Comment:
The previous implementation was easier to read and understand. Please revert this.

Reply:
Shame :)